backend/src/main/java/de/amos/apachepulsarui/controller/TenantController.java

```diff
@@ -25,13 +27,13 @@ public class TenantController {
     private final TenantService tenantService;
 
     @GetMapping()
-    public ResponseEntity<TenantDto> getTenantDetails(@RequestParam String tenantName) {
+    public ResponseEntity<TenantDetailsDto> getTenantDetails(@RequestParam String tenantName) {
         return new ResponseEntity<>(tenantService.getTenantDetails(tenantName), HttpStatus.OK);
     }
 
     @GetMapping("/all")
-    public ResponseEntity<TenantsDto> getAll() {
-        return new ResponseEntity<>(new TenantsDto(tenantService.getAllNames()), HttpStatus.OK);
+    public ResponseEntity<TenantsDto> getAllFiltered(@RequestParam (required = false) List<String> tenants) {
```

Comment:
Maybe use `defaultValue = ""`,  then you dont have to do the null check. This will initialise it as an empty list.